src/PostTypes/WordpressPost.php

```diff
@@ -985,7 +985,7 @@ public static function fetchById($ids) {
 	public static function fetchBySlug($slug){
 		return static::fetchOne(array(
 			'name' => $slug,
-			'post_type' => static::postType(),
+			'post_type' => static::getSelfPostTypeConstraint(),
```

Comment:
This is redundant, as it is done in `fetchOne`
(as is the following line, as `numberposts` is synonymous with `posts_per_page`)